offer-manager-domain/src/main/groovy/life/qbic/business/offers/create/update/UpdateOffer.groovy

```diff
@@ -64,12 +60,23 @@ class UpdateOffer implements CreateOfferInput, CalculatePrice{
                 .build()
 
         if(oldOffer.checksum() != finalizedOffer.checksum()){
+            //todo remove the store method here to avoid duplicate storing?
             storeOffer(finalizedOffer)
         }else{
             output.failNotification("An unchanged offer cannot be updated")
         }
     }
 
+    private life.qbic.datamodel.dtos.business.Offer getOfferById(life.qbic.datamodel.dtos.business.OfferId offerId){
+        Optional<life.qbic.datamodel.dtos.business.Offer> offer = dataSource.getOffer(offerId)
+
+        if(offer.isPresent()) {
+            return offer.get()
+        } else {
+            throw new RuntimeException("No offer is currently selected.")
```

Comment:
We want to call the on error method of the output interface and log the event here maybe